mtags-interfaces/src/main/java/scala/meta/pc/PresentationCompiler.java

```diff
@@ -167,7 +168,7 @@ public abstract CompletableFuture<List<TextEdit>> convertToNamedArguments(Offset
 	/**
 	 * Returns decorations for missing type adnotations, inferred type parameters, implicit parameters and conversions.
 	 */
-	public CompletableFuture<List<SyntheticDecoration>> syntheticDecorations(SyntheticDecorationsParams params) {
```

Comment:
This will break mima, if we have an older mtags that is no longer release this will throw an exception. 

We could do two things:
- have a way to translate older syntheticDecorations to inlay hints (even if worse or not complete) <- this would be the best option
- just deprecate it and have inlay hints return nothing as a default like now (this will make the decorations stop working when using older Scala versions)

Anyway I would not remove the old method

Reply:
> have a way to translate older syntheticDecorations to inlay hints (even if worse or not complete) <- this would be the best option

How should we approach this? In `Compilers.inlayHints`, if `pc.inlayHints` returns empty results, try to run `pc.syntheticDecorations` and transform the result to inlayHints?

Reply:
Actually, we've added `syntheticDecorations` to the  Presentation Compiler in the last release, so the only scala versions for which decorations would stop working are these for which we drop support in the next release (assuming in would contain `inlayHints`)?

Reply:
We also have to take into account the compiler itself, which uses the same interface currently. So we should at least not remove the method. But we might not bother with translating decorations to inlay hints then.